tests/integration/devfile/cmd_devfile_deploy_test.go

```diff
@@ -138,4 +140,48 @@ var _ = Describe("odo devfile deploy command tests", func() {
 			Expect(td.Properties.CmdProperties[segment.ProjectType]).To(ContainSubstring("nodejs"))
 		})
 	})
+
+	When("using a devfile.yaml containing an Image component with a build context", func() {
```

Comment:
In my understanding, `build-image` is executed every time someone does `odo deploy`. It's just invisible to the user, and we have extracted `odo build-image` as a standalone command to provide users the ability to build images without having to leave the odo CLI.

In that case, when build-image is executed every time for deploy logic, does having the same test separately for `odo deploy` add value? I'm wondering because integration tests take a good deal of time to finish, and duplication would only cause it to take more time.

Reply:
I understand and agree with reducing the testing time whenever possible.
But as I saw in the code, the paths used by `odo build-images` and `odo deploy` seemed slightly different, even if they end up calling the same `buildPushImage` function in the `image` package.
This, to me, could justify having separate tests.
If for any reason, we have an issue with `odo deploy` no longer building images, we might want to detect it, no?